app/src/main/java/com/bilginyuksel/sdk/push/NotificationSender.java

```diff
@@ -3,60 +3,41 @@
 import android.app.Notification;
 import android.app.NotificationChannel;
 import android.app.NotificationManager;
-import android.app.PendingIntent;
 import android.content.Context;
-import android.content.Intent;
-import android.os.Build;
 
-import androidx.annotation.RequiresApi;
-import androidx.core.app.NotificationCompat;
+import androidx.annotation.Nullable;
 
+import com.bilginyuksel.sdk.push.utils.Util;
+
+import java.util.concurrent.atomic.AtomicInteger;
 
 public class NotificationSender {
+    private final static String CHANNEL_ID = "push-notification-channel";
+    private final static String CHANNEL_NAME = "push-notification-channel";
 
     private final Context context;
-    private final static String CHANNEL_ID = "push-notification-channel";
+    private final static AtomicInteger ATOMIC_INTEGER = new AtomicInteger();
     private NotificationChannel notificationChannel;
 
     public NotificationSender(Context context) {
-        this.context= context;
-        checkAndroidVersionIsBiggerThanOreo();
-    }
-
-    public void checkAndroidVersionIsBiggerThanOreo(){
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-             notificationChannel = createChannel();
-        }
-    }
-
-    @RequiresApi(api = Build.VERSION_CODES.O)
-    public NotificationChannel createChannel(){
-        return new NotificationChannel(CHANNEL_ID,"push-notification-channel",NotificationManager.IMPORTANCE_HIGH);
+            this.context = context;
+            if (Util.isVersionBiggerThanOreo()) {
+                notificationChannel = new NotificationChannel(CHANNEL_ID, CHANNEL_NAME, NotificationManager.IMPORTANCE_HIGH);
+            }
     }
 
-    @RequiresApi(api = Build.VERSION_CODES.M)
-    public void send() {
-        NotificationManager notificationManager = context.getSystemService(NotificationManager.class);
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-            notificationManager.createNotificationChannel(notificationChannel);
+    public void send(RemoteMessage remoteMessage, @Nullable NotificationChannel channel) {
```

Comment:
Instead of annotating `NotificationChannel` as `@Nullable` we can make it `Optional<NotificationChannel>`.
After we make it Optional we can use the code below
```java
optionalChannel.ifPresentOrElse(notificationManager::createNotificationChannel,
 () -> notificationManager.createNotificationChannel(notificationChannel);
```
This code would do the same thing between line 33,36.